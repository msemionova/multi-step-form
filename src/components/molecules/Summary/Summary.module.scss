.summary {
  flex: 1 1 100%;
}

.container {
  background-color: var(--color-gray-background);
  border-radius: var(--border-radius-medium);
  padding: 2rem;
  margin-bottom: 1rem;

  .head {
    display: flex;
    justify-content: space-between;
    align-items: center;
    border-bottom: var(--border-secondary);
    padding-bottom: 1.4rem;

    button {
      padding: .8rem 0;
      color: var(--color-gray-text-light);
      text-decoration: underline;
      cursor: pointer;

      &:hover {
        color: var(--color-primary-dark);
      }
    }

    h3 {
      font: var(--typography-input-text);
      text-transform: capitalize;
    }

    span {
      font: var(--typography-input-text);
    }
  }

  .list {
    padding-top: 2rem;

    li {
      display: flex;
      justify-content: space-between;
      font: var(--typography-input-label);

      p {
        color: var(--color-gray-text-light);
      }

      &:not(:last-child) {
        margin-bottom: 2rem;
      }
    }
  }
}

.total {
  display: flex;
  justify-content: space-between;
  padding: 2rem;
  font: var(--typography-input-label);
  color: var(--color-gray-text-light);

  span {
    font: var(--typography-input-text);
    color: var(--color-primary-dark);
  }
}
