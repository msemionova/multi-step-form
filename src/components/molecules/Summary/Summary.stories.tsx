import type { Meta, StoryObj } from '@storybook/react';

import StepTitle from './Summary';

const meta: Meta<typeof StepTitle> = {
  component: StepTitle,
};

export default meta;
type Story = StoryObj<typeof StepTitle>;

export const Default: Story = {
  args: {}
};
