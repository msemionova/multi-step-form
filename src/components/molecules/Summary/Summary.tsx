import { useContext } from 'react';
import { Button } from 'src/components/atoms';
import styles from './Summary.module.scss';
import activeStepContext from 'src/context/activeStepContext';

export default function Summary() {
  const { setActiveStep } = useContext(activeStepContext);

  const dummyData = {
    subscriptionType: 'monthly',
    selectedPlan: 'arcade',
    planPrice: 9,
    selectedAddons: [
      {
        title: 'Online service',
        price: 1
      },
      {
        title: 'Larger storage',
        price: 2
      }
    ]
  };

  const priceEnding = dummyData.subscriptionType === 'monthly' ? 'mo' : 'yr';
  const formatPrice = (price: number) => `$${price}/${priceEnding}`;

  const addonsList = dummyData.selectedAddons.map(addon => {
    return <li key={`summary_item_${Math.random()}`}>
      <p>{addon.title}</p>
      <span>{formatPrice(addon.price)}</span>
    </li>;
  });

  const totalPrice = dummyData.selectedAddons.reduce((sum, item) => {
    return sum + item.price;
  }, dummyData.planPrice);

  return (
    <div className={styles.summary}>
      <div className={styles.container}>
        <div className={styles.head}>
          <div>
            <h3>{`${dummyData.selectedPlan} (${dummyData.subscriptionType})`}</h3>
            <Button
              type='transparent'
              onClick={() => setActiveStep(1)}
            >Change</Button>
          </div>
          <span>{formatPrice(dummyData.planPrice)}</span>
        </div>
        <ul className={styles.list}>
          {addonsList}
        </ul>
      </div>
      <div className={styles.total}>
        <p>Total (per {dummyData.subscriptionType === 'monthly' ? 'month' : 'year'})</p>
        <span>{formatPrice(totalPrice)}</span>
      </div>
    </div>
  );
}
