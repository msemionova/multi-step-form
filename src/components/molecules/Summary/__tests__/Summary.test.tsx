import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import Summary from '../Summary';

describe('<Summary />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Summary />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
