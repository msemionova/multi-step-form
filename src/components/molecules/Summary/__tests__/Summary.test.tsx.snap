// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary /> renders correctly 1`] = `
<div
  className="summary"
>
  <div
    className="container"
  >
    <div
      className="head"
    >
      <div>
        <h3>
          arcade (monthly)
        </h3>
        <button
          aria-label=""
          className="button button--transparent"
          data-testid=""
          id=""
          onClick={[Function]}
        >
          <span>
            Change
          </span>
        </button>
      </div>
      <span>
        $9/mo
      </span>
    </div>
    <ul
      className="list"
    >
      <li>
        <p>
          Online service
        </p>
        <span>
          $1/mo
        </span>
      </li>
      <li>
        <p>
          Larger storage
        </p>
        <span>
          $2/mo
        </span>
      </li>
    </ul>
  </div>
  <div
    className="total"
  >
    <p>
      Total (per 
      month
      )
    </p>
    <span>
      $12/mo
    </span>
  </div>
</div>
`;
